cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/Executor.java

```diff
@@ -288,6 +295,59 @@ public void run() {
     }
   }
 
+  /**
+   * A runnable class to auto-adjust the allowed inter-broker partition reassignment concurrency for ongoing executions
+   * using selected broker metrics and based on additive-increase/multiplicative-decrease (AIMD) feedback control algorithm.
+   * Skips concurrency adjustment for demote operations.
+   */
+  private class ConcurrencyAdjuster implements Runnable {
+    private final int _maxPartitionMovementsPerBroker;
+    private LoadMonitor _loadMonitor;
+
+    public ConcurrencyAdjuster() {
+      _maxPartitionMovementsPerBroker = _config.getInt(ExecutorConfig.CONCURRENCY_ADJUSTER_MAX_PARTITION_MOVEMENTS_PER_BROKER_CONFIG);
+      _loadMonitor = null;
+    }
+
+    /**
+     * Initialize the inter-broker partition reassignment concurrency adjustment with the load monitor and the initially
+     * requested inter-broker partition reassignment concurrency.
+     *
+     * @param loadMonitor Load monitor.
+     * @param requestedInterBrokerPartitionMovementConcurrency The maximum number of concurrent inter-broker partition movements
+     *                                                         per broker(if null, use num.concurrent.partition.movements.per.broker).
+     */
+    public synchronized void initAdjustment(LoadMonitor loadMonitor, Integer requestedInterBrokerPartitionMovementConcurrency) {
+      _loadMonitor = loadMonitor;
+      setRequestedInterBrokerPartitionMovementConcurrency(requestedInterBrokerPartitionMovementConcurrency);
```

Comment:
`setRequestedInterBrokerPartitionMovementConcurrency` is a public method in the Executor class. Can it be called elsewhere? My concern is that we only synchronize the methods in ConcurrencyAdjuster but they are calling methods in the outer class. If the outer class method is invoked without going through ConcurrencyAdjuster, there still can be a race condition.

Reply:
The public method `Executor#setRequestedInterBrokerPartitionMovementConcurrency` can be called to handle `Admin` endpoint requests, which just calls the `synchronized` `ExecutionTaskManager#setRequestedInterBrokerPartitionMovementConcurrency` method to update the concurrency. Why is this a concern -- or is there some other race condition that I am missing?

Reply:
Oh I see. I think the missing part is I don't know `ExecutionTaskManager#setRequestedInterBrokerPartitionMovementConcurrency` is already synchronized. Thanks for the explanation. I don't have a concern then.